Apps/W1/Shopify/test/Logs/ShpfySkippedRecordLogTest.Codeunit.al

```diff
@@ -0,0 +1,838 @@
+        SkippedRecord.SetRange("Record ID", SalesInvoiceHeader.RecordId);
+        LibraryAssert.IsTrue(SkippedRecord.FindLast(), 'Skipped record is not created');
+        LibraryAssert.AreEqual(StrSubstNo('Payment terms %1 do not exist in Shopify.', PaymentTermsCode), SkippedRecord."Skipped Reason", 'Skipped reason is not as expected');
+    end;
+
+    [Test]
+    procedure UnitTestLogSalesInvoiceWithCustomerNoIsDefaultCustomerNo()
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        Customer: Record Customer;
+        SkippedRecord: Record "Shpfy Skipped Record";
+        Shop2: Record "Shpfy Shop";
+        PostedInvoiceExport: Codeunit "Shpfy Posted Invoice Export";
+        PaymentTermsCode: Code[10];
+    begin
+        // [SCENARIO] Log skipped record when sales invoice export is skipped because bill to customer no which is default shopify shop customer no.
+        Initialize();
+
+        // [GIVEN] Customer 
+        CreateRandomCustomer(Customer);
+        // [GIVEN] Shopify Customer
+        CreateShopifyCustomer(Customer);
+        // [GIVEN] Shop with default customer no set.
+        CreateShop(Shop2, Customer."No.");
+        // [GIVEN] Payment Terms Code
+        PaymentTermsCode := CreatePaymentTerms(Shop2.Code);
+        // [GIVEN] Shop with default customer no set.
+        SetCustomerAsDefaultForShop(Shop2, Customer);
+        // [GIVEN] Sales Invoice for default customer no.
+        CreateSalesInvoiceHeader(SalesInvoiceHeader, Shop2."Default Customer No.", PaymentTermsCode);
+
+        // [WHEN] Invoke Shopify Posted Invoice Export
+        PostedInvoiceExport.SetShop(Shop2.Code);
+        PostedInvoiceExport.ExportPostedSalesInvoiceToShopify(SalesInvoiceHeader);
+
+        // [THEN] Related record is created in shopify skipped record table.
+        SkippedRecord.SetRange("Record ID", SalesInvoiceHeader.RecordId);
+        LibraryAssert.IsTrue(SkippedRecord.FindLast(), 'Skipped record is not created');
+        LibraryAssert.AreEqual('Bill-to customer no. is the default customer no. for Shopify shop.', SkippedRecord."Skipped Reason", 'Skipped reason is not as expected');
+    end;
+
+    [Test]
+    procedure UnitTestLogSalesInvoiceWithCustomerNoUsedInShopifyCustomerTemplates()
+    var
+        SalesInvoiceHeader: Record "Sales Invoice Header";
+        Customer: Record Customer;
+        SkippedRecord: Record "Shpfy Skipped Record";
+        Shop2: Record "Shpfy Shop";
+        ShopifyCustomerTemplate: Record "Shpfy Customer Template";
+        PostedInvoiceExport: Codeunit "Shpfy Posted Invoice Export";
+        PaymentTermsCode: Code[10];
+    begin
+        // [SCENARIO] Log skipped record when sales invoice export is skipped because customer no which is used in shopify customer templates.
+        Initialize();
+
+        // [GIVEN] Customer 
+        CreateRandomCustomer(Customer);
+        // [GIVEN]  Shop 
+        CreateShop(Shop2, '');
```

Comment:
It's a bit confusing with these shops that don't use the global shop...

Reply:
Creating separate shop to no have to delete Shopify Customer Templates in other tests checking Sales Invoice Export

Reply:
I'd mention this in the comment then. Given Shop without customer templates or something.